File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-6.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
        private int maxSum = Integer.MIN_VALUE;
        public int maxPathSum(TreeNode root){
                function(root);
                return maxSum;
        }
        public int function(TreeNode node){
            if(node==null) return 0;

            int left_sum = Math.max(0, function(node.left));
            int right_sum = Math.max(0, function(node.right));
            int current_sum = node.val+left_sum+right_sum;
            maxSum= Math.max(maxSum, current_sum);

            return node.val + Math.max(left_sum, right_sum);
        }

    
}
    

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        window = set()
        left=0
        best=0

        for right in range(len(s)):
            ch = s[right]
            while ch in window:
                window.remove(s[left])
                left+=1
            window.add(ch)
            best = max(best, right-left+1)
        
        return best

        